ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -635,12 +632,12 @@ public Node visitSelectSingle(SqlBaseParser.SelectSingleContext context) {
                     dereferenceExpression.getFieldName()
                 )
           )) {
-          alias = Optional.of(
+          alias = Optional.of(replaceDotColon(
```

Comment:
Not sure I follow the motivation for this. The argument for `replaceDotColon` is going to be a string without colons, since it is being constructed from two strings without colon, ie. the `base` and the `fieldName`.

Reply:
`base` is not a string, it's an expression and can include `:`. I also made more changes here after switching to `->`. Let me know if you have more questions.